src/app/util/attribute-storage.cpp

```diff
@@ -183,7 +183,7 @@ EmberAfStatus emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, EmberAfEn
     index = static_cast<uint16_t>(realIndex);
     for (uint16_t i = FIXED_ENDPOINT_COUNT; i < MAX_ENDPOINT_COUNT; i++)
     {
-        if (emAfEndpoints[i].endpoint == id)
+        if (emAfEndpoints[i].endpoint == id && emAfEndpoints[i].endpointType != NULL)
```

Comment:
It might have been better to change the init code in `emberAfEndpointConfigure` to set all the endpoint ids on the non-fixed endpoints to the invalid endpoint id....

Reply:
Raised a PR for this: https://github.com/project-chip/connectedhomeip/pull/13814